### PR TITLE
make time/user metadata nullable

### PR DESCRIPTION
**Pull request by hotzevzl**
See https://vizzuality.atlassian.net/browse/MARXAN-342

### CODE REVIEW

File: api/src/migrations/api/1619711501000-DropNotNullsOnTimeUserMetadataColumns.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+INSERT INTO users
+  (email, password_hash, display_name, is_active)
+VALUES
+  (${dummyUserEmailAddress}, crypt('${v4()}', gen_salt('bf')), 'Default owner for data imported via ETL jobs', false);
+`);
+    await Promise.all(
+      this.tablesToAlter.map(async (table) => {
+        await queryRunner.query(`
+UPDATE ${table}
+  SET created_by = (SELECT id FROM users WHERE email = '${dummyUserEmailAddress}')
```
Comment (kgajowy) [inline api/src/migrations/api/1619711501000-DropNotNullsOnTimeUserMetadataColumns.ts:39]: Shouldn't it set `created_at` and `last_modified_at` as well? (might be default now() as well).

File: api/src/migrations/api/1619711501000-DropNotNullsOnTimeUserMetadataColumns.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    await Promise.all(
+      this.tablesToAlter.map(async (table) => {
+        await queryRunner.query(`
+ALTER TABLE ${table}
+  ALTER COLUMN created_at DROP NOT NULL,
+  ALTER COLUMN last_modified_at DROP NOT NULL,
+  ALTER COLUMN created_by DROP NOT NULL;
+`);
+      }),
+    );
```
Comment (kgajowy) [inline api/src/migrations/api/1619711501000-DropNotNullsOnTimeUserMetadataColumns.ts:18]: [minor/style]
```
await Promise.all(
      this.tablesToAlter.map((table) =>
        queryRunner.query(`
ALTER TABLE ${table}
ALTER COLUMN created_at DROP NOT NULL,
ALTER COLUMN last_modified_at DROP NOT NULL,
ALTER COLUMN created_by DROP NOT NULL;`),
      ),
    );
```

File: api/src/migrations/api/1619711501000-DropNotNullsOnTimeUserMetadataColumns.ts (hunk 1 above)
Comment (hotzevzl) [reply api/src/migrations/api/1619711501000-DropNotNullsOnTimeUserMetadataColumns.ts:39]: yep, the down side of this migration doesn't make fully sense in any case as it makes up data (owner and timestamps) that we cannot have if we haven't been gathering them - timestamps will default to `now()`. I don't envisage this side of the migration to have more than a formal value - if needing to roll back, we won't have SQL errors, but owner/timestamps data may be arbitrary.

Comment (kgajowy) [reply api/src/migrations/api/1619711501000-DropNotNullsOnTimeUserMetadataColumns.ts:39]: all good, I wasn't sure if we have `default` set 👍

**Review by kgajowy — APPROVED** [review]

File: api/src/migrations/api/1619711501000-DropNotNullsOnTimeUserMetadataColumns.ts (hunk 2 above)
Comment (hotzevzl) [reply api/src/migrations/api/1619711501000-DropNotNullsOnTimeUserMetadataColumns.ts]: done in ac8f10379e509cce7ad6a530755ad19711d5e2e9

**Pull request merged**